src/server/lib/renderer.ts

```diff
@@ -23,12 +25,20 @@ interface RendererOpts {
 
 const { gaUID } = getConfiguration();
 
-export const renderer: (url: string, opts?: RendererOpts) => string = (
-  url,
-  opts = {},
-) => {
+export const renderer: (
+  url: string,
+  request: Request,
```

Comment:
Rather than pass the Request in every `renderer` call, it's worth adding it to the `RendererOpts` as an optional value, and then only passing in the `request` in when we need the `browserName` on the global state on a particular route.

Reply:
Even better would be to add parse and add the browserName to the globalState only on the route that its needed directly, and have the logic for getting the user agent/brower in its own lib file that can be called.

Reply:
if we make it a property on the global state, is there any advantage to only setting it for specific routes? 
The overhead for setting it is negligible

Reply:
yeah it's negligible, imo it would provide more clarity to whats in the globalState, and opt in as needed

the main concern I have at the moment is that i feel that the renderer should not need the whole request object to render the html, it should only rely on things we pass into it in it's options parameter